### PR TITLE
ec2_instance: add AWSRetry.jittered_backoff on client.describe_instance_attribute

### PR DESCRIPTION
**Pull request by tchernomax**
##### SUMMARY

try to fix InvalidInstanceID.NotFound

> Could not describe attribute groupSet for instance i-…: An error occurred (InvalidInstanceID.NotFound) when calling the DescribeInstanceAttribute operation: The instance ID 'i-…' does not exist

The instance actually exists, it's just an API problem.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

ec2_instance

### CODE REVIEW

**Review by hakbailey — CHANGES_REQUESTED** [review]
@tchernomax thank you for submitting this! The PR looks good but it needs a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#community-changelogs), would you be able to add that?

File: plugins/modules/ec2_instance.py [hunk 1]
```diff
@@ -1456,7 +1462,13 @@ def value_wrapper(v):
 
     if params.get("security_group") or params.get("security_groups"):
         try:
-            value = client.describe_instance_attribute(aws_retry=True, Attribute="groupSet", InstanceId=id_)
+            value = AWSRetry.jittered_backoff(
```
Comment (alinabuzachis) [inline plugins/modules/ec2_instance.py:1465]: I might be wrong, but adding `InvalidInstanceID.NotFound` https://github.com/ansible-collections/amazon.aws/pull/1650/files#diff-a6b28ca5e2f98a5237ad2e88e544b3df3de7bbc95c8dc9088cae572ff92c2f60R2179 should do the job.

**Comment by tchernomax** [comment]
> @tchernomax thank you for submitting this! The PR looks good but it needs a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#community-changelogs), would you be able to add that?

done

File: changelogs/fragments/1650-fix-invalidinstanceid-notfound.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  ec2_instance - retry API call if we get `InvalidGroup.NotFound` error.
```
Comment (hakbailey) [inline changelogs/fragments/1650-fix-invalidinstanceid-notfound.yml:3]:
```suggestion
  ec2_instance - retry API call if we get `InvalidGroup.NotFound` error (https://github.com/ansible-collections/amazon.aws/pull/1650).
```

Comment (tchernomax) [reply changelogs/fragments/1650-fix-invalidinstanceid-notfound.yml]: done

**Review by alinabuzachis — APPROVED** [review]

**Review by hakbailey — APPROVED** [review]

**Comment by hakbailey** [comment]
recheck

**Comment by tchernomax** [comment]
any news ?
Is there something I need to do to make this PR merged ?

**Comment by gravesm** [comment]
recheck

**Comment by tchernomax** [comment]
@gravesm it seems it fail at a different test at each run.

File: plugins/modules/ec2_instance.py [hunk 3]
```diff
@@ -1950,12 +1950,7 @@ def ensure_present(existing_matches, desired_module_state, current_count=None):
         for ins in instances:
             # Wait for instances to exist (don't check state)
             try:
-                AWSRetry.jittered_backoff(
-                    catch_extra_error_codes=["InvalidInstanceID.NotFound"],
-                )(client.describe_instance_status)(
-                    InstanceIds=[ins["InstanceId"]],
-                    IncludeAllInstances=True,
-                )
+                client.describe_instance_status(InstanceIds=[ins["InstanceId"]], IncludeAllInstances=True)
```
Comment (tremble) [inline plugins/modules/ec2_instance.py:1953]: Please leave this line as it was:

1. It's currently explicitly retrying on "InvalidInstanceID.NotFound" to deal with certain race conditions within AWS (it returns the ID, but if you quickly query that API about the ID it fails to find it)
2. Without adding `aws_retry=True` to the describe_instance_status arguments the decorator won't perform any retries.

In this case I'd leave the change you made at line 2159, but use the original code for this call, and add  `InvalidInstanceID.NotFound` to the catch_extra_error_codes parameter that was here.

**Review by tremble — CHANGES_REQUESTED** [review]

**Review by tremble — APPROVED** [review]

**Comment by tremble** [comment]
Thanks for submitting this PR.  I'm sorry it's taken so long to get this merged.

**Pull request merged**